guides/Other Subjects/CreatingPointsSystems.md

```diff
@@ -0,0 +1,161 @@
+async function init() {
+	if (!this.client.gateways.users.schema.has('experience')) {
+		this.client.gateways.users.schema.add('experience', {
+			type: 'integer',
+			default: 0,
+			configurable: false
+		});
+	}
+}
+```
+
+Where we'd check if the schema has the key `experience`. If it doesn't, we add it as a new key, with type `integer` (doubtfully we'll use `float` on this) and make it unconfigurable for the built-in userconf command so the end users do not cheat by modifying their stats.
+
+> **Note**: The name of the key can be any, can be `xp`, `points`... but we will use this one for this guide.
+
+## Setting the monitor
+
+Now that we have set up the schema, we will want to create a monitor
+
+```javascript
+const { Monitor } = require('klasa');
+
+module.exports = class extends Monitor {
+
+	constructor(...args) {
+		super(...args, {
+			enabled: true,
+			ignoreBots: true,
+			ignoreSelf: true,
+			ignoreOthers: false
+		});
+	}
+
+	async run(msg) {
+		// If the message was not sent in a TextChannel, ignore it.
+		if (!msg.guild) return;
+
+		// Update the user's configuration entry by adding 1 to it.
+		await msg.author.configs.update('experience', msg.author.configs.experience + 1);
+	}
+
+};
+
+```
+
+Alternatively, we can create the `init` method and ensure the users' schema always has our key.
+
+## Level up!
+
+Some social bots have level up messages, but, how do we set it up? There are two ways to achieve this:
+
+1. We calculate the current level and the next level on the fly. This system is, however, harder to implement and it processes a lot of maths, but it's also RAM friendly for massive bots. We won't cover this in this guide.
+1. We add a level field. This makes the configuration update slower by nature as it will need to update two values. First, we will create the key:
+
+```javascript
+async function init() {
+	if (!this.client.gateways.users.schema.has('level')) {
+		this.client.gateways.users.schema.add('level', {
+			type: 'integer',
+			default: 0,
+			configurable: false
+		});
+	}
+}
+```
+
+Then we pick up a level calculation algorithm, the following as an example:
+
+```javascript
+Math.floor(0.1 * Math.sqrt(POINTS + 1));
+```
+
+Then inside our monitor's run method:
+
+```javascript
+class extends Monitor {
+
+	// Constructor
+
+	async run(msg) {
+		// If the message was not sent in a TextChannel, ignore it.
+		if (!msg.guild) return;
+
+		// Calculate the next value for experience.
+		const nextValue = msg.author.configs.experience + 1;
+
+		// Cache the current level.
+		const currLevel = msg.author.configs.level;
+
+		// Calculate the next level.
+		const nextLevel = Math.floor(0.1 * Math.sqrt(nextValue + 1));
+
+		// Update the user's configuration entry by adding 1 to it, and update the level aswell.
+		await msg.author.configs.update(['experience', 'level'], [nextValue, nextLevel]);
+
+		// If the current level and the next level do not equal, then it has increased, and you can send the message.
```

Comment:
"If the current level and the next level do not equal" might be better to change to "If the current level and the next level are not the same"